### DISCUSSION TITLE
How to upgrade smarty v2 register_resource function to v3/v4 registerResource function call

### DISCUSSION THREAD

**Discussion by subramanya-n** [smarty-php/smarty#833] [Q&A] [answered]
Hi,
 i'm trying to upgrade smarty version from v3(exact version is v2) to v4. Facing issue while converting register_resource function. 

Before with v2/v3 we were calling using below function
$smarty->register_resource('template', array(
            'smarty_resource_template_source',
            'smarty_resource_template_timestamp',
            'smarty_resource_template_secure',
            'smarty_resource_template_trusted'));
Where as this is deprecated in V4 and also call to function array is deprecated.  
We have some functionality associated with _timestamp and _source function. If we need to achieve the same with V4 registerResource() funciton what we need to change and where we need to add our functionality which is present in _source function.

Can someone help me to understand and help us to update.

Thanks,
Subramanya

**Comment by wisskid** [comment, marked as answer]
You need to wrap your functions in classes that extend Smarty_Resource and register them aa instantiated objects one by one.
